### ISSUE TITLE
Please implement better error messages.

### ISSUE THREAD

**Issue by Asday** [lydell/eslint-plugin-simple-import-sort#10]
```
/home/asday/code/src/app/containers/Navigation.js
  1:1  error  Run autofix to sort these imports!  simple-import-sort/sort
```

Is this complaining about the gap between the parent imports and the sibling imports?  The fact I have `react-navigation` after `react-navigation-redux-helpers`?  Who knows!

It's lovely that you _can_ autofix it, but I don't want you to.  I want to be told what I wrote that was wrong, so I can learn to write the right thing, first time, in future.  That way, when I come to other code on the project that's compliant, I already know what I'm expecting to see, because that's how I've taught myself to write.

**Comment by paulmelnikow** [comment]
I have to say that what I like about this project is that it doesn’t bog you down with incomprehensible details in the messages about how your imports should be arranged. Especially on open-source projects, it saves contributors time to tell them to autofix rather than provide specific errors about what is wrong, which encourages them to spend time on a fairly low value activity of carefully rearranging their imports. It’s great that computers can do this for us.

To train yourself, you could stage your changes, run the autofix, and examine the diff.

**Comment by lydell** [comment]
Yes, like @paulmelnikow said, the idea is to use autofix and forget about it.

> I want to be told what I wrote that was wrong

I don't! I think it's annoying. (When it comes to arbitrary formatting stuff.)

> so I can learn to write the right thing

First off, there's no right way of sorting imports :) Secondly, why waste time on learning and thinking about sorting imports when the computer does it for you?

Previously, I've used other ESLint import sorting rules that gave messages like `"lodash" should come before "react"`. That's no good! I'm not sure there _is_ a good way to provide messages for sorting. That other rule also supported autofix. I saw new developers in my team painstakingly correcting the sorting order one by one until ESLint finally stopped complaining, not realizing they could have autofixed it. I don't want to encourage people to manually fix the sorting, but to nudge them into the workflow of stopping to think about them.

The top reason I created this plugin was to _stop thinking about imports._ I don't think sorted imports make much difference in code comprehensability.

Thanks for your feedback! But I won't implement detailed messages, because it's not worth the complexity of doing so – and it might, in my opinion, even make the plugin worse!

**Comment by Asday** [comment]
>there's no right way of sorting imports
There is - the way the linter has been set up.  If I go put all my imports in reverse alphabetical order on a project I get placed on at work, and the convention is forwards alphabetical order, I have done it incorrectly.

By installing this plugin, I'm defining a correct way.

>Secondly, why waste time on learning and thinking about sorting imports when the computer does it for you?
I already said.

>That way, when I come to other code on the project that's compliant, I already know what I'm expecting to see, because that's how I've taught myself to write.

Clearly you won't be convinced.  Would you be able to give me some pointers on how I could implement these changes myself in a fork?

**Comment by lydell** [comment]
> Clearly you won't be convinced. Would you be able to give me some pointers on how I could implement these changes myself in a fork?

Unfortunately I don't. I avoided solving that problem :) If you do make a fork then please share it here!
